receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper.go

```diff
@@ -104,6 +105,19 @@ func (s *scraper) start(context.Context, component.Host) error {
 func (s *scraper) scrape(ctx context.Context) (pmetric.Metrics, error) {
 	var errs scrapererror.ScrapeErrors
 
+	// If the boot time cache featuregate is disabled, this will refresh the
+	// cached boot time value for use in the current scrape. This functionally
+	// replicates the previous functionality in all but the most extreme
+	// cases of boot time changing in the middle of a scrape.
+	if !bootTimeCacheFeaturegate.IsEnabled() {
+		host.EnableBootTimeCache(false)
+		_, err := host.BootTimeWithContext(ctx)
+		if err != nil {
+			errs.AddPartial(1, fmt.Errorf(`retrieving boot time failed with error "%w", using cached boot time`, err))
+		}
```

Comment:
I'm till not sure why we need it. I'd prefer to keep the Alpha state as it is right now

Reply:
Do you mean keeping the alpha state of the featuregate as the old behaviour of not caching anything?

Reply:
right

Reply:
The reason I don't want to do that is because I want to make it so that with the featuregate on or off, the performance is still down to a more normal level. The code here ends up with what is essentially the same setup as before, but reading the boot time once as opposed to (number of processes) * 4. This has basically the same performance benefit as caching the boot time for the whole run of the process, but maintains functionally the same old behaviour. I mentioned in the comment on the issue that I was going to have the caching turned on by default, and when the featuregate was disabled it would do this. Should I invert it so the featuregate is alpha and we use this behaviour instead by default?

Reply:
I don't like that one feature gate controls two behaviors even if one of them doesn't affect the user experience. But I don't think it's worth debating too much. So we can merge it in this state

Reply:
Please leave a note here and in docs what is needed as a user by cfg or as a developer by "ocb build" to achieve the minimum cpu  load by max caching, because i already feel a bit lost  how will be able to make use of it based on this PR

Reply:
@cforce Nothing will be needed in config or build to get the maximum caching. Caching the boot time for the entire runtime of the collector will be the new default.

Reply:
I want to build prerelase of 0.99 with ocb , which XXXX version/tag/branch name shall i use?
- gomod: github.com/open-telemetry/opentelemetry-collector-contrib/receiver/hostmetricsreceiver XXXX

Reply:
When I need to get a commit-specific version, I go to the component I care about and get the latest commit hash, then run `go list -m <module>@<commit>` which gives me the module pseudo-version.

```
$ go list -m github.com/open-telemetry/opentelemetry-collector-contrib/receiver/hostmetricsreceiver@fcf1bee
github.com/open-telemetry/opentelemetry-collector-contrib/receiver/hostmetricsreceiver v0.98.1-0.20240416201607-fcf1bee48590
```

So based on that in ocb you'd do:
```
- gomod: github.com/open-telemetry/opentelemetry-collector-contrib/receiver/hostmetricsreceiver v0.98.1-0.20240416201607-fcf1bee48590
```

I've never tried mixing versions between components in an ocb config; you might need to use that version for everything if I'm not mistaken.